janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -523,11 +528,11 @@ private void commitDeletes(String collectionName, List<String> deleteIds) throws
     }
 
     @Override
-    public List<String> query(IndexQuery query, KeyInformation.IndexRetriever informations, BaseTransaction tx) throws BackendException {
-        List<String> result;
+    public Stream<String> query(IndexQuery query, KeyInformation.IndexRetriever informations, BaseTransaction tx) throws BackendException {
         String collection = query.getStore();
         String keyIdField = getKeyFieldId(collection);
         SolrQuery solrQuery = new SolrQuery("*:*");
+        solrQuery.set(CommonParams.FL, keyIdField);
```

Comment:
What's this update for?

Reply:
To not ask more information to Solr. With this line, Solr will just return the document id and not all the document. I did the same for ES, I add `"fields" : []` to each request.